src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        /// <remarks>
+        /// The dictionary we get from the VS editorconfig API uses the same dictionary object if there are no changes, so we can cache based on dictionary
+        /// </remarks>
+        private static readonly ConditionalWeakTable<IReadOnlyDictionary<string, object>, SerializableNamingStylePreferencesInfo> _cache = new ConditionalWeakTable<IReadOnlyDictionary<string, object>, SerializableNamingStylePreferencesInfo>();
+        private static readonly object _cacheLock = new object();
+
+        public static SerializableNamingStylePreferencesInfo GetNamingStylesStringFromDictionary(IReadOnlyDictionary<string, object> allRawConventions)
+        {
+            if (!_cache.TryGetValue(allRawConventions, out var value))
+            {
+                lock (_cacheLock)
+                {
+                    if (!_cache.TryGetValue(allRawConventions, out value))
+                    {
+                        value = ParseDictionary(allRawConventions);
+                        _cache.Add(allRawConventions, value);
```

Comment:
[GetOrCreateValue](https://msdn.microsoft.com/en-us/library/dd384250(v=vs.110).aspx) calls the default constructor when the value is not found. @jasonmalinowski  I don't see how this would work?